### PR TITLE
Added Exception Handling and Logging

### PR DESCRIPTION
**Pull request by sritambehera**
1. Phone number field changed into string data type
2. Added Exception Handling
3. Added Logging

### CODE REVIEW

File: TransporterAPI/src/main/java/com/springboot/TransporterAPI/Controller/TransporterController.java [hunk 1]
```diff
@@ -15,50 +19,57 @@
...
+
 	@GetMapping("/transporter")
-	public List<Transporter> getTransporters(@RequestParam(required = false) Boolean transporterApproved,
+	public ResponseEntity<List<Transporter>> getTransporters(
+			@RequestParam(required = false) Boolean transporterApproved,
 			@RequestParam(required = false) Boolean companyApproved,
 			@RequestParam(required = false) Integer pageNo){
-		return service.getTransporters(transporterApproved, companyApproved, pageNo);
+		log.info("Get with Params Controller Started");
+		return new ResponseEntity<>(service.getTransporters(transporterApproved, companyApproved, pageNo),HttpStatus.FOUND);
```
Comment (uckamdar59) [inline TransporterAPI/src/main/java/com/springboot/TransporterAPI/Controller/TransporterController.java:51]: Change FOUND to OK everywhere; in all other services also

File: TransporterAPI/src/main/java/com/springboot/TransporterAPI/Services/TransporterServiceImpl.java [hunk 2]
```diff
@@ -156,109 +101,94 @@ public Transporter getOneTransporter(String transporterId) {
...
-			deleteResponse.setMessage(CommonConstants.accountNotExist);
-			return deleteResponse;
-		}
-		 
+		if(T.isEmpty())
+			throw new EntityNotFoundException(Transporter.class, "id", transporterId.toString());
+
+		transporterdao.deleteById(transporterId);
+		log.info("transporter is deleted in the database");
 	}
```
Comment (uckamdar59) [inline TransporterAPI/src/main/java/com/springboot/TransporterAPI/Services/TransporterServiceImpl.java:192]: In other apis, delete response is sent like status: Deleted successfully, here its not there, pls confirm whether to keep it or not.

File: TransporterAPI/src/main/java/com/springboot/TransporterAPI/Entity/Transporter.java [hunk 3]
```diff
@@ -13,15 +16,18 @@
 public class Transporter {
 	@Id
 	private String transporterId;
-	private Long phoneNo;
+	@Column(unique=true)
+	@NotBlank(message = "Phone no. cannot be blank!")
+	@Pattern(regexp="(^$|[0-9]{10})", message="must be a 10 digit number")
```
Comment (uckamdar59) [inline TransporterAPI/src/main/java/com/springboot/TransporterAPI/Entity/Transporter.java:21]: Also check, random number is not inputted, so check first number is between 6-9

File: TransporterAPI/src/main/java/com/springboot/TransporterAPI/Services/TransporterServiceImpl.java (hunk 2 above)
Comment (sritambehera) [reply TransporterAPI/src/main/java/com/springboot/TransporterAPI/Services/TransporterServiceImpl.java:192]: this status is sent from controller class.

File: TransporterAPI/src/main/java/com/springboot/TransporterAPI/Entity/Transporter.java (hunk 3 above)
Comment (sritambehera) [reply TransporterAPI/src/main/java/com/springboot/TransporterAPI/Entity/Transporter.java:21]: Yes, I will add that into the code.

File: TransporterAPI/src/main/java/com/springboot/TransporterAPI/Controller/TransporterController.java (hunk 1 above)
Comment (sritambehera) [reply TransporterAPI/src/main/java/com/springboot/TransporterAPI/Controller/TransporterController.java:51]: ok

File: TransporterAPI/src/main/java/com/springboot/TransporterAPI/Entity/Transporter.java [hunk 4]
```diff
@@ -13,15 +16,19 @@
...
+	//	"^[6-9]\\d{9}$", "(^$|[0-9]{10})"
+	@Pattern(regexp="(^[6-9]\\d{9}$)", message="Please enter a valid mobile number") 
+	private String phoneNo;
 	private String transporterName;
 	private String companyName;
 	private String transporterLocation;
 	private String kyc;
 	private boolean companyApproved;
 	private boolean transporterApproved;
-
```
[the remark this answers is not in the document]
Comment (sritambehera) [reply TransporterAPI/src/main/java/com/springboot/TransporterAPI/Entity/Transporter.java:23]: Done.

**Review by liveasy97 — APPROVED** [review]

**Pull request merged**